crates/math/src/field/fields/binary/field.rs

```diff
@@ -0,0 +1,611 @@
+        self.value == 0
+    }
+
+    /// Returns true if this element is one
+    #[inline]
+    pub fn is_one(&self) -> bool {
+        self.value == 1
+    }
+
+    /// Returns the underlying value
+    #[inline]
+    pub fn value(&self) -> u128 {
+        self.value
+    }
+
+    /// Returns level number in the tower.
+    #[inline]
+    pub fn num_level(&self) -> usize {
+        self.num_level
+    }
+
+    /// Returns the number of bits needed for that level (2^num_levels).
+    /// Note that the order of the extension field in that level is 2^num_bits.
+    #[inline]
+    pub fn num_bits(&self) -> usize {
+        1 << self.num_level()
+    }
+
+    // Equality check
+    pub fn equals(&self, other: &Self) -> bool {
+        self.value == other.value()
+    }
+
+    /// Returns binary string representation
+    #[cfg(feature = "std")]
+    pub fn to_binary_string(&self) -> String {
+        format!("{:0width$b}", self.value, width = self.num_bits())
+    }
+
+    /// Splits element into high and low parts.
+    /// For example, if a = xy + y + x, then a = (x + 1)y + x and
+    /// therefore, a_hi = x + 1 and a_lo = x.
+    pub fn split(&self) -> (Self, Self) {
+        let half_bits = self.num_bits() / 2;
+        let mask = (1 << half_bits) - 1;
+        let lo = self.value() & mask;
+        let hi = (self.value() >> half_bits) & mask;
+
+        (
+            Self::new(hi, self.num_level() - 1),
+            Self::new(lo, self.num_level() - 1),
+        )
+    }
+
+    /// Joins the hi and low part making a new element of a bigger level.
+    /// For example, if a_hi = x and a_low = 1
+    /// then a = xy + 1.
+    pub fn join(&self, low: &Self) -> Self {
+        let joined = (self.value() << self.num_bits()) | low.value();
+        Self::new(joined, self.num_level() + 1)
+    }
+
+    // It embeds an element in an extension changing the level number.
+    pub fn extend_num_level(&mut self, new_level: usize) {
+        if self.num_level() < new_level {
+            self.num_level = new_level;
+        }
+    }
+
+    /// Create a zero element
+    pub fn zero() -> Self {
+        Self::new(0, 0)
+    }
+
+    /// Create a one element
+    pub fn one() -> Self {
+        Self::new(1, 0)
+    }
+
+    /// Addition between elements of same or different levels.
+    fn add_elements(&self, other: &Self) -> Self {
+        match self.num_level().cmp(&other.num_level()) {
+            Ordering::Greater => {
+                let mask = (1 << other.num_bits()) - 1;
+                // Lsb of a: We get the last "b.num_bits" bits of a.
+                let low = self.value() & mask;
+                // Perform the addition (XOR) on the lower part.
+                let result_low = low ^ other.value();
+                // Msb of a: We get the remaining bits.
+                let high = self.value() >> other.num_bits();
+                // We concatenate both parts.
+                let result_value = (high << other.num_bits()) | result_low;
```

Comment:
I'm a bit confused about this. Wouldn't the high bits of `other` be `0` already, making this essentially the same as `XOR` of both?